.github/workflows/gosec.yaml

```diff
@@ -0,0 +1,27 @@
+---
+name: Go Test on Pull Requests
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    paths:
+      - '**.go'
+  workflow_dispatch:
+jobs:
+  gosec:
+    name: Check GO security
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out code
```

Comment:
One last question then I am good to approve. This seems like the same as https://github.com/redhat-appstudio/release-service/blob/main/.github/workflows/pr.yml#L19
However we are using different fetch-depths and this one specifies the ref (maybe this is the default since the linked one seems to work just fine?) Is there a reason fetch-depth:1 is better than 0? I just think consistency would be ideal here since as I understand it, it is the same action to do the same purpose

Reply:
It's an optimization for speed. 
I don't think `gosec` needs to have the entire git history, just the most recent HEAD of the given branch in the PR.

Reply:
gotcha. makes sense to me then. maybe we should change the `pr.yaml` depth at some point but that is out of scope of this work IMO